dev/AppNotifications/AppNotificationBuilder/AppNotificationProgressBar.cpp

```diff
@@ -0,0 +1,106 @@
+         m_statusStatus{ BindMode::NotSet },
+         m_valueStatus{ BindMode::NotSet },
+         m_valueStringOverrideStatus{ BindMode::NotSet }
+    {};
+
+    void AppNotificationProgressBar::Title(winrt::hstring const& value)
+    {
+        m_title = value;
+        m_titleStatus = BindMode::Value;
+    }
+
+    void AppNotificationProgressBar::Status(winrt::hstring const& value)
+    {
+        m_status = value;
+        m_statusStatus = BindMode::Value;
+    }
+
+    void AppNotificationProgressBar::Value(double value)
+    {
+        THROW_HR_IF_MSG(E_INVALIDARG, value < 0.0 || value > 1.0, "You must provide a value between 0.0 and 1.0");
+
+        m_value = value; m_valueStatus = BindMode::Value;
+    }
+
+    void AppNotificationProgressBar::ValueStringOverride(winrt::hstring const& value)
+    {
+        m_valueStringOverride = value;
+        m_valueStringOverrideStatus = BindMode::Value;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::SetTitle(winrt::hstring const& value)
+    {
+        Title(value);
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::BindTitle()
+    {
+        m_titleStatus = BindMode::Bind;
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::SetStatus(winrt::hstring const& value)
+    {
+        Status(value);
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::BindStatus()
+    {
+        m_statusStatus = BindMode::Bind;
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::SetValue(double value)
+    {
+        Value(value);
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::BindValue()
+    {
+        m_valueStatus = BindMode::Bind;
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::SetValueStringOverride(winrt::hstring const& value)
+    {
+        ValueStringOverride(value);
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::BindValueStringOverride()
+    {
+        m_valueStringOverrideStatus = BindMode::Bind;
+
+        return *this;
+    }
+
+    winrt::hstring AppNotificationProgressBar::ToString()
+    {
+        return wil::str_printf<std::wstring>(L"<progress%ls%ls%ls%ls/>",
+            m_titleStatus == BindMode::NotSet ? L"" : wil::str_printf<std::wstring>(L" title='%ls'", m_titleStatus == BindMode::Value ? m_title.c_str() : L"{progressTitle}").c_str(),
```

Comment:
This is a little complicated to read. Can we break this up by assigning each string first and directly passing the arguments to printf rather than doing everything here?

Reply:
yup

Reply:
For example, we expect 4 strings here as an argument. So lets initialize the 4 strings and break it up before passing it to printf

Reply:
done